x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRankRetrieverBuilder.java

```diff
@@ -44,6 +44,7 @@ public class TextSimilarityRankRetrieverBuilder extends CompoundRetrieverBuilder
     public static final ParseField INFERENCE_ID_FIELD = new ParseField("inference_id");
     public static final ParseField INFERENCE_TEXT_FIELD = new ParseField("inference_text");
     public static final ParseField FIELD_FIELD = new ParseField("field");
+    public static final ParseField LENIENT_FIELD = new ParseField("lenient");
```

Comment:
I'm not sure 'lenient' is quite right - it will try to run it, but if it can't, it will carry on. 'Optional' doesn't quite have the right semantics... maybe 'noncompulsory'?

Reply:
"allow_rank_failures" maybe?

Reply:
This is specific to ML reranking - `allow_rerank_failures`? Or `allow_inference_failures`?

Reply:
`rerank` is good. 

I am not sure we want to say "inference" failures. Let's keep it broad, maybe something fails due to another reason (idk what it could be...).